includes/Helpers.php

```diff
@@ -16,65 +16,133 @@ class Helpers {
 	 * @author Allen Moore, 10up
 	 */
 	public function __construct() {
-		add_action( 'inline_svg', array( $this, 'inline_svg' ), 10, 1 );
+		add_action( 'inline_svg', array( $this, 'inline_svg' ), 10 );
 		add_action( 'svg_button', array( $this, 'svg_button' ), 10, 3 );
 	}
 
 	/**
-	 * Function that returns a SVG file path.
+	 * Function to return the full path to SVG assets.
+	 *
+	 * @author Allen Moore, 10up
+	 * @param  string $path a directory to append to the path.
+	 * @return string       the full path for assets.
+	 */
+	public function get_assets_path( $path = '' ) {
+
+		$full_path = get_svg_path_option();
+
+		/**
+		 * Filters the path.
+		 *
+		 * This function allows the path to include a custom directory
+		 * structure for use cases where SVG assets are stored in multiple
+		 * directories.
+		 *
+		 * @author Allen Moore, 10up
+		 * @param  string $full_path the path to assets from options.
+		 * @param  string $path      a directory to append to the path.
+		 */
+		$full_path = apply_filters( 'svg_path', $full_path, $path );
+
+		return $full_path;
+	}
+
+	/**
+	 * Function to detect the file path type.
+	 *
+	 * @author Allen Moore, 10up
+	 * @return Boolean returns true/false based on the file path type.
+	 */
+	public function get_path_type() {
+
+		$option_path = get_svg_path_option();
+
+		return (bool) filter_var( $option_path, FILTER_VALIDATE_URL	);
+	}
+
+	/**
+	 * Function that returns a file path.
 	 *
 	 * @author Allen Moore, 10up
 	 * @return string the constructed svg path.
 	 */
-	public function get_svg_path() {
+	public function get_path() {
 
 		$theme_path = trailingslashit( get_template_directory() );
-		$svg_path_option = get_svg_path_option();
-		$svg_path = trailingslashit( $theme_path . $svg_path_option );
+		$path = $this->get_assets_path();
+		$svg_path = trailingslashit( $path . $option_path );
 
 		return $svg_path;
 	}
 
 	/**
-	 * Function that returns a SVG file url.
+	 * Function that returns a file url.
 	 *
 	 * @author Allen Moore, 10up
 	 * @param  string $svg the name of the svg file to return.
 	 * @return string      the constructed svg file url.
 	 */
-	public function get_svg_file_path( $svg ) {
+	public function get_file_path( $svg ) {
 
 		if ( empty( $svg ) ) {
 			return;
 		}
-		$svg_path = $this->get_svg_path();
+		$svg_path = $this->get_path();
 
 		$svg_file_path = $svg_path . $svg . '.svg';
 
 		return $svg_file_path;
 	}
 
 	/**
+	 * Function to return the path for a local file.
+	 *
+	 * @author Allen Moore, 10up
+	 * @return string the path for local files.
+	 */
+	public function get_local_file( $svg ) {
+
+		$svg_file = $this->get_file_path( $svg );
+
+		return $svg_file;
+	}
+
+	/**
+	 * Function to return a remote file url.
+	 *
+	 * @author Allen Moore, 10up
+	 * @return string the url for remote files.
+	 */
+	public function get_remote_file( $svg ) {
+
+		$path = $this->get_assets_path();
+		$svg_file = $path . $svg . '.svg';
+
+		return $svg_file;
+	}
```

Comment:
This method looks like it's doing basically the same thing as `get_local_file()` above. Am I missing something?

This is:

1. get the assets path
1. add the svg name to that
1. add .svg to that

get_local_file is:

1. get file path which:
  1. gets the asset path
  1. appends the svg name to it
  1. appends .svg to that

Right?